### ISSUE TITLE
Visual loading model error

### ISSUE THREAD

**Issue by mouxinyue1** [pengzhiliang/MAE-pytorch#94]
Hello, when I was carrying out the visualization, the loading model failed. Could you please tell me the reason?
![image](https://user-images.githubusercontent.com/91049826/202608388-8e4d2a40-91c2-43f1-9c08-61d2d25b523b.png)
I fine-tune the model with pre-training, visualizing calls for fine-tuning the model
![image](https://user-images.githubusercontent.com/91049826/202608490-b9d0508e-f56a-47e0-850d-7014e732ad8c.png)

**Comment by pengzhiliang** [comment]
Take it easy.

Maybe you should load the pretraining model weight for visualizing, not the finetuning weight.

**Comment by pengzhiliang** [comment]
Take it easy.

Maybe you should load the pretraining model weight for visualizing, not the finetuning weight.

**Comment by mouxinyue1** [comment]
The weight I pre-trained can be loaded successfully, but the fine-tuned weight cannot be loaded. The weight I loaded is checkpoint-best-pth

牟新月
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "pengzhiliang/MAE-pytorch"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年11月18日(星期五) 下午4:36
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [pengzhiliang/MAE-pytorch] Visual loading model error (Issue #94)





 
Take it easy.
 
Maybe you should load the pretraining model weight for visualizing, not the finetuning weight.

**Comment by pengzhiliang** [comment]
OK.

For your personal demands, you need to change the tensor name in checkpoint to match its counterpart in the model used in the visualization code.

Hope it can help you.

**Comment by mouxinyue1** [comment]
Hello, can you describe it specifically? Because I don't have a problem with the steps that I'm working on, so I don't know why this error occurs.




牟新月
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "pengzhiliang/MAE-pytorch"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年11月18日(星期五) 下午4:45
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [pengzhiliang/MAE-pytorch] Visual loading model error (Issue #94)





 
OK.
 
For your personal demands, you need to change the tensor name in checkpoint to match its counterpart in the model used in the visualization code.
 
Hope it can help you.

**Comment by mouxinyue1** [comment]
Sorry, I misunderstood, should my visualization be loaded with pre-trained weights? So what does fine tuning do

**Comment by pengzhiliang** [comment]
Yeah. 

As you know, MAE propose to reconstruct pixel values in the pre-training stage. So you can perform visualization to check it by loading the pre-trained model weight.

Masked image modeling aims to provide the favorable initialization of various downstream tasks. So, fine-tuning the pre-training model on downstream tasks (not limit to image classification on ImageNet-1k, but I only implement the classification in this repo) is a quantitive and straight method to assess the quality of pre-training.

Hope it can help you.

**Comment by mouxinyue1** [comment]
I see. Thank you very much for your answer




牟新月
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "pengzhiliang/MAE-pytorch"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年11月18日(星期五) 晚上9:28
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [pengzhiliang/MAE-pytorch] Visual loading model error (Issue #94)





 
Yeah.
 
As you know, MAE propose to reconstruct pixel values in the pre-training stage. So you can perform visualization to check it by loading the pre-trained model weight.
 
Masked image modeling aims to provide the favorable initialization of various downstream tasks. So, fine-tuning the pre-training model on downstream tasks (not limit to image classification on ImageNet-1k, but I only implement the classification in this repo) is a quantitive and straight method to assess the quality of pre-training.
 
Hope it can help you.

**Comment by pengzhiliang** [comment]
You're welcome.
